src/App.tsx

```diff
@@ -1,27 +1,109 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import 'bulma/css/bulma.min.css';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import {
+  getAll,
+  get5First,
+  getRedGoods,
+  wait,
+} from './api/goods';
+import { LoadingError } from './components/LoadingError';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [loadingFirstFive, setLoadingFirstFive] = useState(false);
+  const [loadingRed, setLoadingRed] = useState(false);
+  const [hasError, setHasError] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleAllGoods = async () => {
+    setLoading(true);
+    setHasError(false);
+    await wait(1000);
+    setLoading(false);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    getAll()
+      .then(setGoods)
+      .catch(() => {
+        setLoading(false);
+        setHasError(true);
+      });
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const handle5FirstGoods = async () => {
+    setLoadingFirstFive(true);
+    setHasError(false);
+    await wait(1000);
+    setLoadingFirstFive(false);
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    get5First()
+      .then(setGoods)
+      .catch(() => {
+        setLoadingFirstFive(false);
+        setHasError(true);
+      });
+  };
+
+  const handleRedGoods = async () => {
+    setLoadingRed(true);
+    setHasError(false);
+    await wait(1000);
+    setLoadingRed(false);
+
+    getRedGoods()
+      .then(setGoods)
+      .catch(() => {
+        setLoadingRed(false);
+        setHasError(true);
+      });
+  };
```

Comment:
All these functions are similar, let's instead create one function that will handle loading and pass loading function callback to it